### PR TITLE
FireEye Central Management pack

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/14653
conf-test pr: https://github.com/demisto/content-test-conf/pull/1239

## Description
a whole new pack with fetch

## Minimum version of Cortex XSOAR
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests - UT + TPB + fetch UT
- [x] Documentation  - after cr
- [x] confluence https://confluence.paloaltonetworks.com/display/DemistoContent/FireEye+Central+Management

## TODO:
- [x] outputs to all cmd
- [x] ut
- [x] test fetch/re fetch
- [x] check if we need to add classifiers and incident types - WIP of security team.
- [x] release/delete quarantine email - currently api not working. pending vendor support

### CODE REVIEW

**Comment by bakatzir** [comment]
```
2021-05-18 14:18:28.4138 info (FireEye Central Management_instance_1_FireEye Central Management_fetch-incidents) FireEye Central Management no new alerts were fetched at: {'last_alert_ids': '["35019", "35021", "35022", "35024", "35026", "35028", "35030", "35032", "35033", "35035", "35037", "35039"]', 'time': '2021-05-11 22:53:29 +0000'}
```

File: Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+def fetch_incidents(client: Client, last_run: dict, first_fetch: str, max_fetch: int = 50,
+                    info_level: str = 'concise') -> Tuple[dict, list]:
+    if not last_run:  # if first time fetching
+        next_run = {
+            'time': to_fe_datetime_converter(first_fetch),
+            'last_alert_ids': []
+        }
+    else:
+        next_run = last_run
+
```
Comment (yuvalbenshalom) [inline Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py:487]: please add log for current next run to be used.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good, see my comments

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

File: Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+    raw_response = client.get_alerts_request(request_params={
+        'start_time': to_fe_datetime_converter(first_fetch),
+        'info_level': info_level
+    })
+    all_alerts = raw_response.get('alert')
+
+    if not all_alerts:
+        demisto.info(f'{INTEGRATION_NAME} no alerts were fetched at: {str(next_run)}')
+        # as no alerts occurred till now, update last_run time accordingly
+        last_run['time'] = to_fe_datetime_converter('now')
```
Comment (yuvalbenshalom) [inline Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py:486]: why do you change and return last_run if you copied it in the beginning to next_run?

File: Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+                'rawJSON': json.dumps(alert)
+            }
+            incidents.append(incident)
+            last_alert_ids.append(alert_id)
+
+    if not incidents:
+        demisto.info(f'{INTEGRATION_NAME} no new alerts were fetched at: {str(next_run)}')
+        # as no alerts occurred till now, update last_run time accordingly
+        last_run['time'] = to_fe_datetime_converter('now')
+        return last_run, []
```
Comment (yuvalbenshalom) [inline Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py:511]: again, I think you should use next run

File: Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+    demisto.info(f'{INTEGRATION_NAME} executing fetch with: {str(next_run.get("time"))}')
+
+    raw_response = client.get_alerts_request(request_params={
+        'start_time': to_fe_datetime_converter(first_fetch),
+        'info_level': info_level
+    })
+    all_alerts = raw_response.get('alert')
+
+    if not all_alerts:
+        demisto.info(f'{INTEGRATION_NAME} no alerts were fetched at: {str(next_run)}')
```
Comment (yuvalbenshalom) [inline Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py:484]:
```suggestion
        demisto.info(f'{INTEGRATION_NAME} no alerts were fetched at: {str(last_run)}')
```

File: Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+
+        return self._http_request(method='GET',
+                                  url_suffix='reports/report',
+                                  params=params,
+                                  resp_type='content',
+                                  timeout=timeout)
+
+
+@logger
+def to_fe_datetime_converter(time_given: str = 'now') -> str:
```
Comment (yuvalbenshalom) [inline Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py:138]: BTW, are we using this on other integrations also?

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
impressive work with the fields.
see my comments.

File: Packs/CommonTypes/IncidentFields/incidentfield-Alert_Acknowledgement.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  "system": false,
+  "systemAssociatedTypes": null,
+  "threshold": 72,
+  "type": "shortText",
+  "unmapped": false,
+  "unsearchable": false,
+  "useAsKpi": false,
+  "validationRegex": "",
+  "version": -1,
+  "fromVersion": "5.0.0"
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/CommonTypes/IncidentFields/incidentfield-Alert_Acknowledgement.json:40]: for customers that uses the NX pack(fromversion 5.0.0) and would like to use the new mapper with the new fields.

File: Packs/CommonTypes/IncidentFields/incidentfield-SHA256.json [hunk 7]
```diff
@@ -1,37 +1,37 @@
...
-	"systemAssociatedTypes": null,
-	"threshold": 72,
-	"type": "shortText",
-	"unmapped": false,
-	"unsearchable": false,
-	"useAsKpi": true,
-	"validationRegex": "",
-	"version": -1,
-	"fromVersion": "5.0.0"
+  "associatedToAll": true,
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/CommonTypes/IncidentFields/incidentfield-SHA256.json]: no, just auto indent.

File: Packs/CommonTypes/IncidentFields/incidentfield-Signature.json [hunk 8]
```diff
@@ -1,38 +1,38 @@
...
- ],
- "systemAssociatedTypes": [
-  "Vulnerability",
-  "Malware"
- ],
- "associatedToAll": false,
- "unmapped": false,
- "unsearchable": false,
- "fromVersion": "5.0.0"
+  "id": "incident_signature",
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/CommonTypes/IncidentFields/incidentfield-Signature.json:2]: indeed, just auto indent.

File: Packs/FireEyeEX/README.md [hunk 9]
```diff
@@ -0,0 +1 @@
+<!-- Currently holds the incident type for FE CM, will be edited when an integration will be added -->
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/FireEyeEX/README.md:1]: this pack will have an actual readme real with this: https://github.com/demisto/content/pull/13041

File: Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py (hunk 5 above)
Comment (bakatzir) [reply Packs/FireEyeCM/Integrations/FireEyeCM/FireEyeCM.py:138]: not yet. will be used in https://github.com/demisto/content/pull/13041, will consolidate to an APIModule.

File: Packs/CommonTypes/IncidentFields/incidentfield-Incident_Link.json [hunk 10]
```diff
@@ -17,10 +17,7 @@
     "content": true,
     "group": 0,
     "hidden": false,
-    "associatedTypes": [
-        "FraudWatch Incident"
-    ],
-    "associatedToAll": false,
+    "associatedToAll": true,
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Packs/CommonTypes/IncidentFields/incidentfield-Incident_Link.json:20]: yes, per @melamedbn mapping revamp.

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**